geode-gfsh/src/main/java/org/apache/geode/management/internal/cli/commands/ListGatewayCommand.java

```diff
@@ -53,9 +57,23 @@ public ResultModel listGateway(
           help = CliStrings.LIST_GATEWAY__MEMBER__HELP) String[] onMember,
       @CliOption(key = {CliStrings.GROUP, CliStrings.GROUPS},
           optionContext = ConverterHint.MEMBERGROUP,
-          help = CliStrings.LIST_GATEWAY__GROUP__HELP) String[] onGroup)
+          help = CliStrings.LIST_GATEWAY__GROUP__HELP) String[] onGroup,
+      @CliOption(key = {CliStrings.LIST_GATEWAY__SHOW_RECEIVERS_ONLY},
+          specifiedDefaultValue = "true",
+          help = CliStrings.LIST_GATEWAY__SHOW_RECEIVERS_ONLY__HELP) Boolean showReceiversOnly,
```

Comment:
You can use boolean instead Boolean, if you put unspecifiedDefaultVaue="false"

Reply:
I need Boolean to have three possible values: true, false and null. I need the null value for the interceptor check.

Reply:
in the interceptors, you can error out if both parameters have "true" value. You don't need to check for null

Reply:
if you use `unspecifiedDefaultVaue="false"` then how could you differentiate between `list gateways --receivers-only=false --senders-only=false` and `list gateways --senders-only=false`? In both cases, the interceptor would receive `showOnlyReceivers=false` and `showOnlySenders=false` but the first command should return error and the second one should work.

Reply:
I might be confused, but to me 
`list gateways`
`list gateways --sends-only=false`
`list gateways --receivers-only=false
`list gateways --receivers-only=false --senders-only=false` 

should all behave the same, since "false" is the default value for these two options. In each case, it all means " I want to see all of them"

Reply:
you forgot the extra restriction that both options cannot be used together, so the fourth command you wrote will return an error :)

Reply:
that's only when both values are true at the same time. The only invalid command should be:
`list gateways --receivers-only=true--senders-only=true` or
`list gateways --receivers-only --senders-only`

when either or both values are false, there should not be an error.

Reply:
I think that using both parameters at the same time has no sense at all, and if someone tries to use them, it will be due to a misunderstanding of the documentation, this is why I like the idea of returning an error if both parameters are introduced.

Reply:
I think we are disagreeing on what should be the outcome of the following command
`list gateways --receivers-only=false --senders-only=false`
My take is we should go ahead and show all of them, and you would like to throw an error. Probably we should involve some doc's people to interpret this.

Reply:
yes, I want to return an error due to I want to return an error whenever both options are used together. @davebarnes97 , could you help with this?

Reply:
I understand you're asking me as a writer to help with the description of the behavior, but I think there's an inherent difficulty in the design. So 2 answers, first the design level, then the language level.
There are three choices of what to display: all, receivers, senders. So this is not a binary/boolean choice, it's ternary. The proposed implementation is a pair of booleans giving 4 total choices, where one is forbidden. I would like to suggest, instead, a parameter that specifies what to show, and recognizes 3 choices. For example,
  list gateways --display=senders
  list gateways --display=receivers
  list gateways --display=all
A value other than (senders, receivers, all) is an error.
If --display is not specified, the default is 'all' (the current behavior).

Reply:
Leaving aside my design recommendation, and focusing only on language, I think the difficulty arises because the choices "--receivers-only" and "--senders-only" are actually negatives (meaning "not senders" and "not receivers, respectively). If you recast them as positives "--receivers" and "--senders", there is no need to assign them true/false values. Presence means true, absence means false. So:
  list gateways --senders      // show senders
  list gateways --receivers   // show receivers
  list gateways                       // show both (current behavior)
  list gateways --senders --receivers  // show both
In this scheme, since specifying both is the same as specifying neither, there is no 4th case to worry about.

Reply:
My 2 cents:
the user will probably ever only care about:
`list gateways`
`list gateways --receivers-only`
`list gateways --senders-only`

Anything else (explicitly appending `=false`, specifying both, etc) falls under "don't care" in my opinion.  You could throw an error in every permutation other than the 3 above, or do anything that feels consistent with principal-of-least-surprise.

@davebarnes97 has a great suggestion to think about maybe constraining the input space another way to avoid the need for so much validation, but it does lead to a slightly "less natural" command syntax imho.

Reply:
@davebarnes97 assuming we want to leave the design as is, let's just focus on the semantic meaning of the following command:
`list gateways --receivers-only=false --senders-only=false`

Is the above command self-conflicting in anyway that we have to throw an error? I know `list gateways --receivers-only --senders-only` is self-conflicting, but with false value, I don't think it is, but it's your call.

Honestly,  it's very unlikely that users will issue such command, so exactly what it does might not be that important, it just introduces some coding complexity that I was just trying to get rid of in the first place.

Reply:
I agree that treating `--<flag>=false` any differently from omitting it entirely violates principal of least surprise.

Reply:
Thanks you all for your contributions. This sentence from @onichols-pivotal changed my mind:

> I agree that treating `--<flag>=false` any differently from omitting it entirely violates principal of least surprise.

If I think about using `false` as if the parameter is omitted, then it has sense for me not returning an error when using `--receivers-only=false --senders-only=false`. Thanks again and sorry @jinmeiliao for the misunderstanding.